ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/P2TargetUtils.java

```diff
@@ -1024,12 +1027,33 @@ static IQueryable<IInstallableUnit> getQueryableMetadata(URI[] repos, IProgressM
 		if (result.size() != repos.length) {
 			throw new CoreException(repoStatus);
 		}
+		result.addAll(additional);
```

Comment:
`additional` could contain references that are already present in `result`. Making result a `LinkedHashSet` would avoid this.

Reply:
I now unify this as the last step.